x-pack/plugin/security/lib/nimbus-jose-jwt-fixed/src/main/java/com/nimbusds/jose/util/InnerJSONObjectUtils.java

```diff
@@ -0,0 +1,515 @@
+/*
+ * nimbus-jose-jwt
+ *
+ * Copyright 2012-2016, Connect2id Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package com.nimbusds.jose.util;
+
+import com.nimbusds.jose.shaded.gson.Gson;
+import com.nimbusds.jose.shaded.gson.GsonBuilder;
+import com.nimbusds.jose.shaded.gson.ToNumberPolicy;
+import com.nimbusds.jose.shaded.gson.internal.LinkedTreeMap;
+import com.nimbusds.jose.shaded.gson.reflect.TypeToken;
+
+import java.lang.reflect.Type;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.text.ParseException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/*
+ * Copied from nimbus-jose-jwt version 9.37.3.
+ *
+ * The only modifications in this file are formatting & warning suppression as necessary to work with our infrastructure, tweaks to
+ * field visibility to enable this wrapping strategy, and this comment.
+ */
+
+/**
+ * JSON object helper methods.
+ *
+ * @author Vladimir Dzhuvinov
+ * @version 2022-08-19
+ */
+@SuppressWarnings({ "unchecked", "rawtypes" })
+public class InnerJSONObjectUtils {
```

Comment:
Rather than copy the source, we should be able to rename the existing classfile when pulling in the dependency?
https://gradleup.com/shadow/configuration/relocation/#filtering-relocation

Reply:
Doesn't shadowJar rewrite the rest of the JAR to "correctly" reference the moved class when doing that? If I've misunderstood on that point please do correct me, I think that would be better it's just not how I understood that part of shadowJar to work.

Reply:
You are right, it does. However, I think we can still workaround that. We essentially want 3 parts to the final jar:
* The original jar minus these two classes
* These two classes renamed
* The two replacement classes you are adding in

The first two bullets can be separate shadowjar tasks, and ultimately everthing from these 3 parts will go into jar.

Reply:
Aha, so essentially several intermediate gradle tasks to build a few temporary JARs with the right properies, then merge them into a single final target? Yeah, I think I see how that can work, let me give it a shot.

Reply:
@rjernst This is done, I agree this is an improvement even though shadowJar gave me a bit of a headache about it. Thanks!